src/dom/rect.js

```diff
@@ -234,21 +247,63 @@ export default class Rect {
 	}
 
 	/**
+	 * Checks if all properties ({@link #top}, {@link #left}, {@link #right},
+	 * {@link #bottom}, {@link #width} and {@link #height}) are the same as in the other `Rect`.
+	 *
+	 * @param {module:utils/dom/rect~Rect} rect A `Rect` instance to compare with.
+	 * @returns {Boolean} `true` when Rects are equal. `false` otherwise.
+	 */
+	isEqual( anotherRect ) {
+		for ( const prop of rectProperties ) {
+			if ( this[ prop ] !== anotherRect[ prop ] ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks whether a `Rect` fully contains another `Rect` instance.
+	 *
+	 * @param {module:utils/dom/rect~Rect} anotherRect
+	 * @returns {Boolean} `true` if contains, `false` otherwise.
+	 */
+	contains( anotherRect ) {
+		const intersectRect = this.getIntersection( anotherRect );
+
+		if ( !intersectRect || !intersectRect.isEqual( anotherRect ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
 	 * Returns a rect of the web browser viewport.
 	 *
 	 * @returns {module:utils/dom/rect~Rect} A viewport rect.
+	 * @param {Boolean} [options.excludeScrollbars] When set `true` the `Rect` will not include
+	 * the scrollbars of the viewport.
 	 */
-	static getViewportRect() {
+	static getViewportRect( options = {} ) {
```

Comment:
Why does this method look like one big hack? There's something wrong with this method's constructor if you need to override stuff like `_source` and call private methods once an instance was initialised.